base/dict.jl

```diff
@@ -714,14 +702,17 @@ length(t::Dict) = t.count
     i == 0 && return nothing
     i = skip_deleted(v.dict, i)
     i == 0 && return nothing
-    vals = T <: KeySet ? v.dict.keys : v.dict.vals
-    (@inbounds vals[i], i == typemax(Int) ? 0 : i+1)
+    if T <: KeySet
+        (@inbounds v.dict.pairs[i].first, i == typemax(Int) ? 0 : i+1)
+    else
+        (@inbounds v.dict.pairs[i].second, i == typemax(Int) ? 0 : i+1)
+    end
```

Comment:
Maybe slightly nicer as:
```suggestion
    p = @inbounds v.dict.pairs[i]
    return p[T <: KeySet ? 1 : 2], i == typemax(Int) ? 0 : i+1
```